Editor/NeedsRuntimeConversion/Extensions/UrdfRobotExtensions.cs

```diff
@@ -63,34 +63,36 @@ public static void Create(string filename, ImportSettings settings)
             Undo.RegisterCreatedObjectUndo(robotGameObject, "Create " + robotGameObject.name);
             Selection.activeObject = robotGameObject;
 
-            CorrectAxis(robotGameObject, settings.choosenAxis);
+            CorrectAxis(robotGameObject);
             CreateCollisionExceptions(robot, robotGameObject);
         }
 
-        public static void CorrectAxis(GameObject robot, ImportSettings.axisType axis = ImportSettings.axisType.yAxis)
+        public static void CorrectAxis(GameObject robot)
         {
-            UrdfVisual[] visualMeshList = robot.GetComponentsInChildren<UrdfVisual>();
-            UrdfCollision[] collisionMeshList = robot.GetComponentsInChildren<UrdfCollision>();
             UrdfRobot robotScript = robot.GetComponent<UrdfRobot>();
+            if (robotScript.CheckOrientation())
+                return;
+            Quaternion correctYtoZ = Quaternion.Euler(-90, 0, 90);
+            Quaternion correctionZtoY = Quaternion.Inverse((correctYtoZ));
+            Quaternion correction = new Quaternion();
```

Comment:
Looks a job for https://docs.unity3d.com/ScriptReference/Quaternion.FromToRotation.html ? 

The code could be rewritten to this simple one-liner:

Quaternion correction = (robotScript.choosenAxis == ImportSettings.axisType.zAxis) ? Quaternion.FromToRotation(Vector3.up, Vector3.forward) : Quaternion.FromToRotation(Vector3.forward, Vector3.up);

Reply:
Yes this is a god way to do it. But just wanted it to be more verbose with variable names so it is easier to read.

Reply:
It might be better to use Unity's recommended way to change the axis?
```
correctYtoZ = Quaternion.FromToRotation(Vector3.up, Vector3.forward);
and correctZtoY = Quaternion.FromToRotation(Vector3.forward, Vector3.up);
```

Reply:
This is not the correct way in my case. This rotation using eulers make sure all three axes are aligned in the direction I need them to. FromTo API produces a rotation which is constraint in just one axes. And it achieves this by using just a single rotation change.